### DISCUSSION TITLE
Weird behavior of `connect` and `packetArrived`

### DISCUSSION THREAD

**Discussion by bongjunj** [ANLAB-KAIST/KENSv3#119] [Project Q&A 2. Transport Layer (TCP)] [answered]
<img width="462" alt="Screen Shot 2022-10-10 at 9 27 28 PM" src="https://user-images.githubusercontent.com/7106939/194866208-e3eaf149-e982-45fb-985b-6ec08dca96c0.png">

<img width="462" alt="Screen Shot 2022-10-10 at 9 27 54 PM" src="https://user-images.githubusercontent.com/7106939/194866288-6714f8ef-3730-43e7-8544-54c8c82f89c0.png">

<img width="798" alt="Screen Shot 2022-10-10 at 9 28 12 PM" src="https://user-images.githubusercontent.com/7106939/194866333-3ed49160-2fd5-4b55-89ce-7a7af389d07e.png">

`connect` has been never called, but `packetArrived` receives `SYN` packet.
How is this possible?

**Comment by dain7672** [comment]
Same situation. Probably it is possible to be related to #117 and #114

**Comment by smpak19** [comment]
I got some information from this https://github.com/ANLAB-KAIST/KENSv3/discussions/72#discussioncomment-2538556
I think test cases starts with 'Test(__)' only use __ from your implementation.

**Reply by smpak19** [reply]
So when you test accept or other functions, your printf in connect may not called.

**Reply by bongjunj** [reply]
Thank you for letting me know.

**Comment by bongjunj** [comment, marked as answer]
When 'accept' implementation is tested, the 'connect' function will be from the solution binary. It means my 'connect' implementation is not called. And similar things happen when 'connect' is tested.
